src/app/core/cache/object-cache.reducer.ts

```diff
@@ -1,17 +1,27 @@
 import {
-  ObjectCacheAction, ObjectCacheActionTypes, AddToObjectCacheAction,
-  RemoveFromObjectCacheAction, ResetObjectCacheTimestampsAction
+  ObjectCacheAction,
+  ObjectCacheActionTypes,
+  AddToObjectCacheAction,
+  RemoveFromObjectCacheAction,
+  ResetObjectCacheTimestampsAction,
+  AddPatchObjectCacheAction, ApplyPatchObjectCacheAction
 } from './object-cache.actions';
-import { hasValue } from '../../shared/empty.util';
+import { hasValue, isNotEmpty } from '../../shared/empty.util';
 import { CacheEntry } from './cache-entry';
 import { ResourceType } from '../shared/resource-type';
+import { applyPatch, Operation } from 'fast-json-patch';
 
 export enum DirtyType {
   Created = 'Created',
   Updated = 'Updated',
   Deleted = 'Deleted'
 }
 
+export interface Patch {
```

Comment:
Please add TypeDocs for this new interface.  (Thanks for adding them for the new functions in this same file!)